python/ray/train/torch/config.py

```diff
@@ -207,4 +209,5 @@ def on_shutdown(self, worker_group: WorkerGroup, backend_config: TorchConfig):
     def on_training_start(
         self, worker_group: WorkerGroup, backend_config: BackendConfig
     ):
+        worker_group.execute(init_torch_device_manager)
```

Comment:
Nit: instead of doing this explicitly, we can lazily instantiate the device manager whenever the first call to `get_torch_device_manager` happens.